crates/bin/output_segment/src/main.rs

```diff
@@ -0,0 +1,82 @@
+use std::fs;
+use std::path::PathBuf;
+
+use cairo_vm::hint_processor::hint_processor_utils::felt_to_usize;
+use cairo_vm::types::layout_name::LayoutName;
+use cairo_vm::vm::runners::cairo_pie::CairoPie;
+use cairo_vm::Felt252;
+use clap::Parser;
+use prove_block::{debug_prove_error, get_memory_segment, prove_block, DEFAULT_COMPILED_OS};
+
+#[derive(Parser, Debug)]
+struct Args {
+    /// Pie-zip to compare.
+    #[arg(long = "pie-path")]
+    zip_pie_path: PathBuf,
+
+    /// RPC endpoint to use for fact fetching
+    #[arg(long = "rpc-provider", default_value = "http://localhost:9545")]
+    rpc_provider: String,
+}
+
+fn init_logging() {
+    env_logger::builder()
+        .filter_level(log::LevelFilter::Debug)
+        .format_timestamp(None)
+        .try_init()
+        .expect("Failed to configure env_logger");
+}
+
+#[tokio::main]
+async fn main() {
+    init_logging();
+
+    let args = Args::parse();
+
+    let zip_pie_path = args.zip_pie_path;
+    let endpoint = args.rpc_provider;
+
+    let reference_pie_bytes = fs::read(zip_pie_path).unwrap();
```

Comment:
Some ideas instead of unwrap:

* `expect()`
* propagate error (make `main()` return some type of `Result`)
* use something like `anyhow`: https://docs.rs/anyhow/latest/anyhow/